koku/masu/util/common.py

```diff
@@ -488,3 +490,43 @@ def __call__(cls, *args, **kwargs):
                 instance = super().__call__(*args, **kwargs)
                 cls._instances[cls] = instance
         return cls._instances[cls]
+
+
+def check_setup_complete(provider_uuid):
+    with ProviderDBAccessor(provider_uuid=provider_uuid) as provider_accessor:
+        return provider_accessor.get_setup_complete()
+
+
+def get_provider_updated_timestamp(provider_uuid):
+    with ProviderDBAccessor(provider_uuid=provider_uuid) as provider_accessor:
+        return provider_accessor.get_data_updated_timestamp()
+
+
+def get_start_delta(start_date, provider_uuid):
+    """Get initial start date for download file processing"""
+    dh = DateHelper()
+    if start_date.year == dh.today.year and start_date.month == dh.today.month:
+        last_update_day = get_provider_updated_timestamp(provider_uuid)
```

Comment:
We might want to re-think using this timestamp to determine our range. Just thinking of this scenario: we don't download data for a provider for say, 5 days, but a customer updated a cost-model on this provider. The data_updated_timestamp would be based on when the cost model was updated and not on when we last downloaded data.

I think we need to set this range based on the data we have, not the last time the data was updated.